packages/autorest.python/autorest/codegen/serializers/utils.py

```diff
@@ -11,3 +12,11 @@ def method_signature_and_response_type_annotation_template(
     response_type_annotation: str,
 ) -> str:
     return f"{method_signature} -> {response_type_annotation}:"
+
+
+def to_snake_case(name: str) -> str:
+    return re.sub(
+        "((?!^)(?<!_)[A-Z][a-z]+|(?<=[a-z0-9])[A-Z])",
+        r"_\1",
+        name.replace("-", "").replace(" ", "_"),
+    ).lower()
```

Comment:
we can't reuse same function of `_util.py` since there are many special example name in swagger. Just try `autorest --version=3.9.2 --python --use=C:\dev\autorest.python\packages\autorest.python\. --use=@autorest\modelerfour@4.24.3 --python-sdks-folder=C:\dev\azure-sdk-for-python\sdk     C:\dev\azure-rest-api-specs\specification\network\resource-manager\readme.md   --version-tolerant=false --generate-sample --include-x-ms-examples-original-file `

Reply:
what are the differences between the two? Is the reason there are differences due to swagger errors etc?

Reply:
there is no restriced rulethe example name in swagger